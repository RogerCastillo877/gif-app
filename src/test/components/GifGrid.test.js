import React from "react";
import { shallow } from "enzyme";
import { GifGrid } from "../../components/GifGrid";
import { useFetchGifs } from "../../hooks/useFetchGifs";
jest.mock("../../hooks/useFetchGifs");

describe('should test GifsGrid component', () => {

    const category = 'samurai x'; 
    test('should show properly', () => {

        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        })

        const wrapper = shallow(<GifGrid category={category} />);

        expect( wrapper ).toMatchSnapshot();
    })

    test('should show items when load img useFetchGifs', () => {

        const gifs = [{
            id: 'ABC',
            url: 'https://img.jpg',
            title: 'Ese es'
        }]
        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        })

        const wrapper = shallow( <GifGrid category={ category } />);
        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('p').exists() ).toBe(false);
        expect( wrapper.find('GifGridItem').length ).toBe( gifs.length );
    })
    
})
