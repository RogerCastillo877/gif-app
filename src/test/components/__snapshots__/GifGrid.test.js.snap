// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should test GifsGrid component should show items when load img useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
     
    samurai x
     
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Ese es"
      url="https://img.jpg"
    />
  </div>
</Fragment>
`;

exports[`should test GifsGrid component should show properly 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
     
    samurai x
     
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
